[Agent Builder] Conversation Rounds re-design

chrisbmar opened the pull request:
## Summary

closes https://github.com/elastic/search-team/issues/11904
closes https://github.com/elastic/search-team/issues/11903

**Included in this PR:**
- Redesign of the conversation_rounds
- We now show all completed steps with a tick icon on the left, if the step is the last step and the round is not yet completed, we show a loading icon (we now use a factory pattern for rendering the steps as we need to count the steps displayed to ensure only the last item has the loading icon)
- You can now click on tool call displays (accordion) and the dropdown content will show the parameters used in that tool call
- Adds an error result step
- File structure update of components within conversation_rounds (this is the reason so many files look changed but reviewing the PR won't be that scary it's just a lot of moved files)
- Fixes an intermittent issues where conversation rounds would not scroll to the top occasionally after onSubmit

**NOT included in this PR:**
- Persisting the error round state on the frontend
- The new loading icon (for now we’re using elastic loading icons) - _will be done in a separate PR_
- on-hover actions at the bottom of each round I.e. copy - _will be done in a separate PR_


https://github.com/user-attachments/assets/51d20364-52b0-4782-8ada-63e861ff857d


https://github.com/user-attachments/assets/43aa3edb-83d6-4450-9f9b-0fb7969f42e6


https://github.com/user-attachments/assets/89e1e3fa-e5a8-4953-877f-a9c54474b0ff

Error result step:
<img width="1728" height="997" alt="image" src="https://github.com/user-attachments/assets/cbeb71ea-1d7d-4482-93cb-66b3f50c731f" />

chrisbmar committed:
move round input to dedicated component

chrisbmar committed:
separate thinking from response message

chrisbmar committed:
update input styles to match designs

chrisbmar committed:
separate thinking panel and thinking title

chrisbmar committed:
thinking panel wip

chrisbmar committed:
remove styles file

chrisbmar committed:
small style polishes

chrisbmar committed:
refactor round steps to use a factory pattern to identify the last item in the column

chrisbmar committed:
extract components and reduce round_steps file length

chrisbmar committed:
white space

chrisbmar committed:
remove flex grow from round containers

chrisbmar committed:
Add error result step and format query result step

chrisbmar committed:
thinking time and view round json display

chrisbmar committed:
small tidy up

chrisbmar committed:
separate round thinking happy path and error state

chrisbmar committed:
folder structure update

chrisbmar committed:
error round styles

chrisbmar committed:
console warning

chrisbmar committed:
show parameters accordion dropdown

chrisbmar committed:
loading icon refactor

chrisbmar committed:
add spacer for accordion content

chrisbmar committed:
dont scroll to top of the round until next render

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

chrisbmar committed:
rounded border on round thinking panel

chrisbmar committed:
fix i18n issue

chrisbmar committed:
more i18n fix

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/365769)
* Commit: 021cc8f56395ddcedff471df82a2735d86e29198

### Failed CI Steps
* [FTR Configs #49](https://buildkite.com/elastic/kibana-pull-request/builds/365769#019ac055-e36d-4339-b367-2b1b2bb8c8cd)
* [FTR Configs #49](https://buildkite.com/elastic/kibana-pull-request/builds/365769#019ac075-490a-4d47-8d54-811d6db1d28a)
* [FTR Configs #75](https://buildkite.com/elastic/kibana-pull-request/builds/365769#019ac055-e38a-4c9d-b575-878a076ae0a2)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/365769#019ac055-e36d-4339-b367-2b1b2bb8c8cd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/365769/jobs/019ac055-e36d-4339-b367-2b1b2bb8c8cd/artifacts/019ac075-22a7-472d-b423-038f3e466419) FTR Configs #<span></span>49 / Agent Builder converse Conversation Flow sends a message with tool call and receives response with thinking
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/365769#019ac075-490a-4d47-8d54-811d6db1d28a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/365769/jobs/019ac075-490a-4d47-8d54-811d6db1d28a/artifacts/019ac07e-3bba-4341-acda-841a076deca5) FTR Configs #<span></span>49 / Agent Builder converse Conversation Flow sends a message with tool call and receives response with thinking
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/365769#019ac055-e38a-4c9d-b575-878a076ae0a2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/365769/jobs/019ac055-e38a-4c9d-b575-878a076ae0a2/artifacts/019ac073-ca65-4252-992e-ab5ea0d73c5f) FTR Configs #<span></span>75 / Visualizations - Group 3 lens app - TSVB Open in Lens Dashboard to TSVB to Lens should convert a by value TSVB viz to a Lens viz

### History
* :broken_heart: [Build #365733](https://buildkite.com/elastic/kibana-pull-request/builds/365733) failed f5ee720bbf39abaf1d61d144b621f35934f06585
* :broken_heart: [Build #365709](https://buildkite.com/elastic/kibana-pull-request/builds/365709) failed 69fa76855e1705c3dde547def92d7857f39e8a75

cc @chrisbmar

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365769","number":365769,"commit":"021cc8f56395ddcedff471df82a2735d86e29198"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365733","number":365733,"commit":"f5ee720bbf39abaf1d61d144b621f35934f06585"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365709","number":365709,"commit":"69fa76855e1705c3dde547def92d7857f39e8a75"}],"number":365769}
buildkite-pr-comment-->

chrisbmar commented:
hey @julianrosado thanks for the feedback let me reply to you here:
1. Can you be more specific where we need to change the padding sorry my brain is not understanding "user response".
2. Ohhh this is an interesting one because actually all those checkmarks (icons) are `alignItems=center` with the text on all lines with the exception of the "Calling tool...." one, that one isn't because it's actually an accordion with an icon to the left of it. I will play around with the styles to see if I can update the accordion to match the others. But to answer your original question, they already should be aligned 😕 I'll see what custom stuff I can add.
<img width="1130" height="373" alt="image" src="https://github.com/user-attachments/assets/49c09642-53be-4993-a762-ac8ff8bf4a9b" />

4. Sure, we could but I'd like us to think a bit more about this to be consistent with how we display errors. Showing 1 error a different way from another seems a bit off to me. The context exceeded error isn't a case that we're hitting a lot right now. So for now, I will leave this as is. But, please do include this as an action item when you're back and we can tackle the error states together? Let me know if that works for you.
5. Sure will make the gutter "m"
6. I guess with some custom code I could make the artifacts appear as aligned with the text above it. But please let's not mistake these artifacts as being direct descendants of the text that appears above it (like a 'header') as this is not true, that relationship does not exist. For now, I will make the error step and the ESQL step appear aligned with the text above it 🫡 
7. Got it. That's on my radar. There's already an eui style for that, I added it and removed it the other day, I'll find it again and add it in a future PR.
8. Let's tackle this when you're back outside of this PR.

Regarding the rest:

> - Thinking text is same size as body, not one size smaller
> - Font looks heavier. What is the weight of the one on the actual site now? I know in figma I am using regular weight, so I am not sure if it's a browser level thing that is rendering it thicker?
> - Space between completed reasoning and the actual response looks bigger than intended? Feels like too much white space.

I will look into these as well, but I think I copied the styles that were in Figma, could be wrong. 

I will let you when I've updated the code and updated the videos/image in this PR 🙏🏽

chrisbmar committed:
julian design review

chrisbmar committed:
fix

chrisbmar committed:
fix FTR tests once and for all

chrisbmar committed:
Merge branch 'main' into agent-builder-reasoning-panel-11903

chrisbmar committed:
fix whitespace

chrisbmar commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/round_thinking/steps/tool_call_display.tsx:
```diff
@@ -0,0 +1,65 @@
+/*
```
No change here, just moved to a dedicated file.

chrisbmar commented on x-pack/platform/plugins/shared/onechat/public/application/hooks/use_conversation_scroll_actions.ts:
```diff
@@ -112,7 +112,9 @@ export const useConversationScrollActions = ({
-    scrollToMostRecentRound({ scrollContainer, position: 'start' });
+    requestAnimationFrame(() => {
+      scrollToMostRecentRound({ scrollContainer, position: 'start' });
+    });
```
Since the conversation input refactor we've needed this I think - and I'm not sure why. Something to do with the ordering of rendering/painting to the DOM. We should **not** be calling `scrollToMostRecentRound` until the next time we paint to the DOM I.e. after the next render. We need the render to complete so that the `conversation_round` has the minimum height of 1 full screen height so that there is space for us to scroll it to the top.

jedrazb commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/round_thinking/steps/error_result_step.tsx:
```diff
@@ -0,0 +1,45 @@
+          </EuiText>
+        </EuiSplitPanel.Inner>
+        <EuiSplitPanel.Inner paddingSize="none">
+          <EuiCodeBlock language="esql" isCopyable paddingSize="m" lineNumbers>
```
Any reason we use `esql` language for displaying error payload?

jedrazb commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/round_thinking/steps/thinking_item_layout.tsx:
```diff
@@ -0,0 +1,115 @@
+            </EuiText>
+          </EuiSplitPanel.Inner>
+          <EuiSplitPanel.Inner paddingSize="none">
+            <EuiCodeBlock language="esql" isCopyable paddingSize="m" lineNumbers>
```
Why `esql` language in code block if we are rendering `json`?

jedrazb commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/round_thinking/round_thinking_panel.tsx:
```diff
@@ -0,0 +1,120 @@
+  defaultMessage: 'Reasoning',
+});
+const completedReasoningLabel = i18n.translate('xpack.onechat.conversation.completedReasoning', {
+  defaultMessage: 'Completed Reasoning',
```
afaik we don't capitailize 2nd word, so should be `Completed reasoning`

jedrazb commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/round_thinking/steps/round_steps.tsx:
```diff
@@ -50,206 +46,23 @@ const flyoutResultTypes = [
-      return <OtherResultStep result={toolResult} />;
+const getItemIcon = (isLastItem: boolean, isLoading: boolean): ReactNode => {
+  if (isLastItem && isLoading) {
+    return <EuiLoadingElastic size="m" aria-label={'Loading...'} />;
```
i18n for aria label?

jedrazb reviewed (commented):
Nice! 🔥  Tested locally and feels great. Couple of small issues/nits in comments

chrisbmar committed:
jedr pr review

chrisbmar commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/round_error/context_exceeded_round_error.tsx:
```diff
@@ -0,0 +1,48 @@
+            values={{
+              docsLink: (
+                <EuiLink
+                  href={`${docLinks.limitationsKnownIssues}#conversation-length-exceeded`}
```
do you mean just the `href` assigned to a variable or the full `values` object assigned to a variable?

chrisbmar replied:
https://github.com/elastic/kibana/pull/244325/commits/abfdd8a5cefdf0d1ce53516f0cad3bece7922486

chrisbmar committed:
joe pr feedback

joemcelroy commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/round_input.tsx:
```diff
@@ -0,0 +1,57 @@
+};
+
+// TODO: migrate to EUI
+const borderRadius = '12px 12px 0 12px';
```
flagging this, might of missed this

chrisbmar committed:
pr feedback

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366475)
* Commit: e63b2ee668801597a4555874667f02fdec04824a

### Failed CI Steps
* [FTR Configs #106](https://buildkite.com/elastic/kibana-pull-request/builds/366475#019ac627-623d-41e7-a4ee-ffc791b10565)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/6a67d59439d5eecf76f067455c818d867d82d879) | [after](https://github.com/elastic/kibana/commit/e63b2ee668801597a4555874667f02fdec04824a) | diff |
| --- | --- | --- | --- |
| `onechat` | 1482 | 1494 | +12 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6a67d59439d5eecf76f067455c818d867d82d879) | [after](https://github.com/elastic/kibana/commit/e63b2ee668801597a4555874667f02fdec04824a) | diff |
| --- | --- | --- | --- |
| `onechat` | 433.6KB | 439.6KB | +6.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/6a67d59439d5eecf76f067455c818d867d82d879) | [after](https://github.com/elastic/kibana/commit/e63b2ee668801597a4555874667f02fdec04824a) | diff |
| --- | --- | --- | --- |
| `onechat` | 21.5KB | 21.7KB | +200.0B |



### History
* :green_heart: [Build #366191](https://buildkite.com/elastic/kibana-pull-request/builds/366191) succeeded 17478dbb0304fbabc13ee4294f8a80c3ba642829
* :broken_heart: [Build #366037](https://buildkite.com/elastic/kibana-pull-request/builds/366037) failed 510a0f046bfa07104d9cf8caa483644a68ea4c87
* :broken_heart: [Build #365769](https://buildkite.com/elastic/kibana-pull-request/builds/365769) failed 021cc8f56395ddcedff471df82a2735d86e29198
* :broken_heart: [Build #365733](https://buildkite.com/elastic/kibana-pull-request/builds/365733) failed f5ee720bbf39abaf1d61d144b621f35934f06585
* :broken_heart: [Build #365709](https://buildkite.com/elastic/kibana-pull-request/builds/365709) failed 69fa76855e1705c3dde547def92d7857f39e8a75

cc @chrisbmar

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366475","number":366475,"commit":"e63b2ee668801597a4555874667f02fdec04824a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366191","number":366191,"commit":"17478dbb0304fbabc13ee4294f8a80c3ba642829"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366037","number":366037,"commit":"510a0f046bfa07104d9cf8caa483644a68ea4c87"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365769","number":365769,"commit":"021cc8f56395ddcedff471df82a2735d86e29198"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365733","number":365733,"commit":"f5ee720bbf39abaf1d61d144b621f35934f06585"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/365709","number":365709,"commit":"69fa76855e1705c3dde547def92d7857f39e8a75"}],"number":366475}
buildkite-pr-comment-->
